spec/expression.dd

```diff
@@ -1100,6 +1100,60 @@ $(GNAME CastExpression):
         (foo) - p;      // subtract p from foo
         -------------
 
+$(H4 $(LNAME2 cast_basic_data_types, Basic Data Types))
+    $(P There are some cases where the $(DDSUBLINK spec/type, implicit-conversions, implicit conversion)
+        cannot be applied, so cast may be used to force the type system to accept the reinterpretation.
+    )
+
+    $(P This situation may occur, for example, when trying to assign a value stored on four bytes
+        to a variable stored on one byte. Without a cast, the following error will be displayed:
+        "cannot implicitly convert expression a of type int to byte".
```

Comment:
Rephrase to: "An example of such a scenario is represented by trying to store a wider type into a narrower one:"